conclave-host/src/main/kotlin/com/r3/conclave/host/internal/NativeEnclaveHandle.kt

```diff
@@ -15,21 +22,57 @@ class NativeEnclaveHandle(
 ) : EnclaveHandle {
     private val enclaveFile: Path
     private val enclaveId: Long
-    override val enclaveInterface: HostEnclaveInterface
+    override val enclaveInterface: NativeHostEnclaveInterface
+    override lateinit var quotingService: EnclaveQuoteService
+
+    /**
+     * Handler for servicing requests from the enclave for signed quotes.
+     */
+    private inner class GetSignedQuoteHandler : CallHandler {
+        override fun handleCall(parameterBuffer: ByteBuffer): ByteBuffer {
+            val report = Cursor.slice(SgxReport, parameterBuffer)
+            val signedQuote = quotingService.retrieveQuote(report)
+            return signedQuote.buffer
+        }
+    }
 
     init {
-        require(enclaveMode != EnclaveMode.MOCK)
+        require(enclaveMode != EnclaveMode.MOCK) {
+            MOCK_MODE_UNSUPPORTED_MESSAGE
+        }
+
         NativeLoader.loadHostLibraries(enclaveMode)
         enclaveFile = Files.createTempFile(enclaveClassName, "signed.so").toAbsolutePath()
         enclaveFileUrl.openStream().use { Files.copy(it, enclaveFile, REPLACE_EXISTING) }
         enclaveId = Native.createEnclave(enclaveFile.toString(), enclaveMode != EnclaveMode.RELEASE)
         enclaveInterface = NativeHostEnclaveInterface(enclaveId)
         NativeApi.registerHostEnclaveInterface(enclaveId, enclaveInterface)
+
+        enclaveInterface.apply {
```

Comment:
Apologies my ignorance, why `apply`?

Reply:
No ignorance! There just used to be more statements in there! Will remove.